projects/plugins/boost/app/assets/src/js/features/ui/collapsible-meta/collapsible-meta.tsx

```diff
@@ -0,0 +1,94 @@
+import styles from './collapsible-meta.module.scss';
+import { Button } from '@automattic/jetpack-components';
+import React, { useState, useEffect } from 'react';
+import ChevronDown from '$svg/chevron-down';
+import ChevronUp from '$svg/chevron-up';
+import { recordBoostEvent } from '$lib/utils/analytics';
+
+type CollapsibleMetaProps = {
+	children: React.ReactNode;
+	header?: React.ReactNode;
+	summary?: React.ReactNode;
+	toggleText: string;
+	headerText?: string;
+	useChevron?: boolean;
+	tracksEvent?: string;
+	extraButtons?: React.ReactNode;
+	onToggleHandler?: ( isExpanded: boolean ) => void;
+};
+
+/*
+ * This component is used to create a collapsible meta section.
+ * This is used by the modules on the settings page to create a more consistent UI.
+ * The toggleText and tracksEvent props are used by every module.
+ * headerText is used by Page Cache and Minify modules.
+ * The Quality Settings module uses the header prop to render a custom header.
+ * The extraButtons prop is used to render a custom set of buttons, for example, the "Clear Cache" button.
+ * The onToggleHandler prop is a callback function that is called when the collapsible meta is toggled. Used by Minify module to reset the input value.
+ */
+const CollapsibleMeta = ( {
+	children,
+	header,
+	summary,
+	toggleText,
+	tracksEvent = '',
+	extraButtons,
+	headerText,
+	onToggleHandler,
+}: CollapsibleMetaProps ) => {
+	const [ isExpanded, setIsExpanded ] = useState( false );
+
+	/*
+	 * A callback function that is called when the collapsible meta is toggled.
+	 * The callback function is passed as a prop to the component and is called with the new state.
+	 */
+	useEffect( () => {
+		onToggleHandler?.( isExpanded );
+	}, [ isExpanded, onToggleHandler ] );
+
+	useEffect( () => {
+		if ( tracksEvent !== '' ) {
+			recordBoostEvent( tracksEvent, {
+				status: isExpanded ? 'open' : 'close',
+			} );
+		}
+	}, [ isExpanded, tracksEvent ] );
```

Comment:
I can't remember who, but someone told me to really try to not use `useEffect` in my components.

I think both of these can be avoided and the code can be simplified by creating a custom function that's called when the button is clicked!

```js
	const onToggle = () => {
		const newIsExpanded = ! isExpanded;
		setIsExpanded( newIsExpanded );
		onToggleHandler?.( newIsExpanded );
		if ( tracksEvent !== '' ) {
			recordBoostEvent( tracksEvent, {
				status: newIsExpanded ? 'open' : 'close',
			} );
		}
	};
```

I tried adding a suggestion for this, but it was too big and it wasn't clear what's being changed.

If you end up using this, don't forget to remove the `useEffect` import.